generator/src/cmd/generate.go

```diff
@@ -18,20 +18,116 @@ package cmd
 import (
 	"application-generator/src/pkg/generate"
 	"github.com/spf13/cobra"
+	"application-generator/src/pkg/model"
+	"bufio"
+	"fmt"
+	"os"
+	"strings"
 	"strconv"
 )
 
+const (
+	SvcMaxNumberDefault = 10
+	SvcReplicaMaxNumberDefault = 10
+	SvcEpMaxNumberDefault = 5
+)
+
+func yesNoPrompt(label string, def bool) bool {
+  choices := "Y/n"
+  if !def {
+      choices = "y/N"
+  }
+
+  r := bufio.NewReader(os.Stdin)
+  var s string
+
+  for {
+      fmt.Fprintf(os.Stderr, "%s (%s) ", label, choices)
+      s, _ = r.ReadString('\n')
+      s = strings.TrimSpace(s)
+      if s == "" {
+          return def
+      }
+      s = strings.ToLower(s)
+      if s == "y" || s == "yes" {
+          return true
+      }
+      if s == "n" || s == "no" {
+          return false
+      }
+  }
+}
+
+func stringPrompt(label string) string {
+  var s string
+  r := bufio.NewReader(os.Stdin)
+  for {
+      fmt.Fprint(os.Stderr, label+" ")
+      s, _ = r.ReadString('\n')
+      if s != "" {
+          break
+      }
+  }
+  return strings.TrimSpace(s)
+}
+
 var generateCmd = &cobra.Command{
-	Use:   "generate [chain-file] [cluster-file] [k8s-readiness-probe]",
-	Short: "This commands generates Kubernetes manifest files by using example files in chains and clusters directory and also uses k8s readiness probe time",
-	Args:  cobra.ExactArgs(2),
+	Use:   "generate [mode] [input-file]",
+	Short: "This commands can be run under two different modes: (i) 'random' mode which generates a random description file or (ii) 'preset' mode which generates Kubernetes manifest based on a description file in the input directory",
+	Args:  cobra.RangeArgs(1, 2),
 	Run: func(cmd *cobra.Command, args []string) {
-		chain := args[0]
-		readinessProbe, err := strconv.Atoi(args[1])
-		exitIfError(err)
 
-		config, clusters := generate.Parse(chain)
-		generate.Create(config, readinessProbe, clusters)
+		mode := args[0]
+
+		var inputFile string
+		if mode == "random" {
+			simpleMode := yesNoPrompt("Do you want to set simple configuration? (otherwise, extended)", true)
+
+			// NOTE: Here we assume numerically consecutive names for clusters and namespaces
+			clusterNamePrefix := stringPrompt("What is your cluster name prefix?")
```

Comment:
What if the cluster names does not have a prefix and does not have any pattern. The similar thing applies to other prefixes as well.

Reply:
In that case, random generation of applications would not be so useful since the user would still need to setup many parameters manually...

Reply:
We should consider that our end-users have already set up their clusters and then they are going to use this tool to generate random architecture.  And most of the time they don't care about different name spaces and they usually use default namespaces. Maybe we should ask how many clusters do you have and then ask their names (most of the time they don't have too much clusters maybe 2 or three) and we could simply bypass different namespaces and generate everything for default name space.